### PR TITLE
Security Features

### PR DESCRIPTION
**Pull request by oremj**
Adds:

* more callback validation
* parameter validation
* namespace validation

@jvehent r?

### CODE REVIEW

File: proxyservice/dockerhub.go [hunk 1]
```diff
@@ -54,8 +56,10 @@ type DockerHubWebhookData struct {
 
 // Callback calls data's callback_url
 func (d *DockerHubWebhookData) Callback(cb *CallBackData) error {
-	if !strings.HasPrefix(d.CallbackURL, "https://registry.hub.docker.com/") {
-		return errors.New("d.CallBackURL does not start with https://registry.hub.docker.com/")
+	callbackPrefix := fmt.Sprintf("%s/u/%s/%s/hook/",
+		DockerhubRegistry, d.Repository.Namespace, d.Repository.Name)
+	if !strings.HasPrefix(d.CallbackURL, callbackPrefix) {
```
Comment (jvehent) [inline proxyservice/dockerhub.go]: At this point, should this be an equal match rather than a prefix check?

File: proxyservice/jenkins.go [hunk 2]
```diff
@@ -94,6 +95,16 @@ func (j *Jenkins) TriggerJob(jobPath string, params url.Values) error {
...
 func (j *Jenkins) TriggerDockerhubJob(data *DockerHubWebhookData) error {
+	if !regexp.MustCompile(`^[a-zA-Z0-9_-]+$`).MatchString(data.Repository.Name) {
+		return fmt.Errorf("Invalid data.Repository.Name: %s", data.Repository.Name)
+	}
+	if !regexp.MustCompile(`^[a-zA-Z0-9_-]+$`).MatchString(data.Repository.Namespace) {
+		return fmt.Errorf("Invalid data.Repository.Namespace: %s", data.Repository.Namespace)
+	}
+	if !regexp.MustCompile(`^[a-zA-Z0-9_-\.]+$`).MatchString(data.PushData.Tag) {
+		return fmt.Errorf("Invalid data.PushData.Tag: %s", data.PushData.Tag)
+	}
```
Comment (jvehent) [inline proxyservice/jenkins.go]: nit: limit length to, say, 100 characters

**Comment by jvehent** [comment]
r+ with comments

File: proxyservice/dockerhub.go (hunk 1 above)
Comment (oremj) [inline proxyservice/dockerhub.go]: The hook contains a random hash after `/hook/`, something like `/hook/2141b5bi5i5b02bec211i4eeih0242eg11000a/`

**Pull request merged**